packages/app-store/hubspotothercalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,212 @@
+  }
+
+  private hubspotContactSearch = async (event: CalendarEvent) => {
+    const publicObjectSearchRequest: PublicObjectSearchRequest = {
+      filterGroups: event.attendees.map((attendee) => ({
+        filters: [
+          {
+            value: attendee.email,
+            propertyName: "email",
+            operator: "EQ",
+          },
+        ],
+      })),
+      sorts: ["hs_object_id"],
+      properties: ["hs_object_id", "email"],
+      limit: 10,
+      after: 0,
+    };
+
+    return await hubspotClient.crm.contacts.searchApi
+      .doSearch(publicObjectSearchRequest)
+      .then((apiResponse) => apiResponse.results);
+  };
+
+  private getHubspotMeetingBody = (event: CalendarEvent): string => {
+    return `<b>${event.organizer.language.translate("invitee_timezone")}:</b> ${
+      event.attendees[0].timeZone
+    }<br><br><b>${event.organizer.language.translate("share_additional_notes")}</b><br>${
+      event.additionalNotes || "-"
+    }`;
+  };
+
+  private hubspotCreateMeeting = async (event: CalendarEvent) => {
+    const simplePublicObjectInput: SimplePublicObjectInput = {
+      properties: {
+        hs_timestamp: Date.now().toString(),
+        hs_meeting_title: event.title,
+        hs_meeting_body: this.getHubspotMeetingBody(event),
+        hs_meeting_location: getLocation(event),
+        hs_meeting_start_time: new Date(event.startTime).toISOString(),
+        hs_meeting_end_time: new Date(event.endTime).toISOString(),
+        hs_meeting_outcome: "SCHEDULED",
+      },
+    };
+
+    return hubspotClient.crm.objects.meetings.basicApi.create(simplePublicObjectInput);
+  };
+
+  private hubspotAssociate = async (meeting: any, contacts: any) => {
+    const batchInputPublicAssociation: BatchInputPublicAssociation = {
+      inputs: contacts.map((contact: any) => ({
+        _from: { id: meeting.id },
+        to: { id: contact.id },
+        type: "meeting_event_to_contact",
+      })),
+    };
+    return hubspotClient.crm.associations.batchApi.create(
+      "meetings",
+      "contacts",
+      batchInputPublicAssociation
+    );
+  };
+
+  private hubspotUpdateMeeting = async (uid: string, event: CalendarEvent) => {
+    const simplePublicObjectInput: SimplePublicObjectInput = {
+      properties: {
+        hs_timestamp: Date.now().toString(),
+        hs_meeting_title: event.title,
+        hs_meeting_body: this.getHubspotMeetingBody(event),
+        hs_meeting_location: getLocation(event),
+        hs_meeting_start_time: new Date(event.startTime).toISOString(),
+        hs_meeting_end_time: new Date(event.endTime).toISOString(),
+        hs_meeting_outcome: "RESCHEDULED",
+      },
+    };
+
+    return hubspotClient.crm.objects.meetings.basicApi.update(uid, simplePublicObjectInput);
+  };
+
+  private hubspotDeleteMeeting = async (uid: string) => {
+    return hubspotClient.crm.objects.meetings.basicApi.archive(uid);
+  };
+
+  private hubspotAuth = (credential: Credential) => {
```

Comment:
This function sets the token within the HubSpot API Client package, so no need to return the actual access token to be used down the path.